src/main/java/nextstep/app/SecurityConfig.java

```diff
@@ -78,26 +66,25 @@ public AuthenticationManager authenticationManager() {
     }
 
     @Bean
-    public SecurityFilterChain securityFilterChain() {
-        return new DefaultSecurityFilterChain(
-                List.of(
-                        new SecurityContextHolderFilter(),
-                        new UsernamePasswordAuthenticationFilter(authenticationManager()),
-                        new BasicAuthenticationFilter(authenticationManager()),
-                        new OAuth2AuthorizationRequestRedirectFilter(clientRegistrationRepository()),
-                        new OAuth2LoginAuthenticationFilter(clientRegistrationRepository(), new OAuth2AuthorizedClientRepository(), authenticationManager()),
-                        new AuthorizationFilter(requestAuthorizationManager())
+    public SecurityFilterChain securityFilterChain2(HttpSecurity http) {
+        return http
+                .authorizeHttpRequests(authorize -> authorize
+                        .requestMatchers("/members").hasRole("ADMIN")
+                        .requestMatchers("/members/me").authenticated()
+                        .anyRequest().permitAll()
```

Comment:
```suggestion
    public SecurityFilterChain securityFilterChain(HttpSecurity http) {
        return http
                .authorizeHttpRequests(authorize -> authorize
                        .requestMatchers("/members").hasRole("ADMIN")
                        .requestMatchers("/members/me").authenticated()
                        .anyRequest().permitAll()
                        .exceptionHandling().authenticationEntryPoint(new CustomAuthenticationEntryPoint())
                )
```

리팩터링의 흔적이 남아있군요 ㅎㅎ
추가로 예외 상황에 따라 어떻게 응답을 할 수 있을까? 라는 고민을 하셨던 만큼 해당 부분이 확장 포인트가 될 수도 있을 것 같네요~

- 메서드 이름에 2 제거
- exceptionHandling 내용 추가

Reply:
재연님 덕분에 나중에 실제 시큐리티 이용해서 비즈니스를 구현할 때도 비교적 어렵지 않게 예외 핸들링을 구현해볼 수 있을 것 같습니다. :)